spectral_fitting/fitting_with_spectralcube.ipynb

```diff
@@ -0,0 +1,697 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Some residuals remain, however, the fit is fairly good.\n",
+    "\n",
+    "One major feature of `astropy.modeling.models` is the built-in handling of units. The fitted model `g_fit` already has each parameter in useful physical units."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Exercise ###\n",
+    "\n",
+    "Can the above be example fit be improved with a more sophisticated model? Try the fit using multiple Gaussian components."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using dask ###\n",
+    "\n",
+    "When using a large spectral-line data cube, the [dask](https://dask.org/) library can be used in spectral-cube for operations instead of numpy. See the [spectral-cube documentation](https://spectral-cube.readthedocs.io/en/latest/dask.html) on dask integration for a thorough description.\n",
+    "\n",
+    "Most of the above code for fitting a spectrum will be unchanged. What will change is how the data cube is read in:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cube = SpectralCube.read(datafile, use_dask=True, format='fits')\n",
+    "cube = cube.with_spectral_unit(u.km / u.s)\n",
+    "\n",
+    "cube.use_dask_scheduler('synchronous')\n",
+    "\n",
+    "from dask.diagnostics import ProgressBar\n",
+    "\n",
+    "pbar = ProgressBar()\n",
+    "pbar.register()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using dask is enabled with `use_dask=True` and the remainder of the code (i) sets using synchronous (serial) operations and (ii) enables printing a progress bar when dask is performing an operation.\n",
+    "\n",
+    "Now when we look at the cube properties:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": "DaskSpectralCube with shape=(300, 64, 64) and unit=K and chunk size (300, 64, 64):\n n_x:     64  type_x: RA---SIN  unit_x: deg    range:    23.408277 deg:   23.428647 deg\n n_y:     64  type_y: DEC--SIN  unit_y: deg    range:    30.750561 deg:   30.768065 deg\n n_s:    300  type_s: VRAD      unit_s: km / s  range:     -280.000 km / s:     -79.670 km / s"
+     },
+     "metadata": {},
+     "execution_count": 12
+    }
+   ],
+   "source": [
+    "cube"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "we see that this is a [DaskSpectralCube](https://spectral-cube.readthedocs.io/en/latest/api/spectral_cube.DaskSpectralCube.html#spectral_cube.DaskSpectralCube) along with additional information. Dask will load the data in chunks, and the size of the chunks for this data cube are shown as the `chunk size`.\n",
```

Comment:
Maybe explain that there is just one chunk because it's a small cube but that for large cubes there would be multiple chunks and this prevents the data from ever being loaded all in one go?

Reply:
Re-ordered this section w/ more explanation of each step